django/db/migrations/writer.py

```diff
@@ -350,6 +350,8 @@ def serialize(cls, value):
         # process.
         if isinstance(value, Promise):
             value = force_text(value)
+        elif isinstance(value, LazyObject):
+            value = value.__reduce__()[1][0]
```

Comment:
Okay, it was just an idea. I didn't have any idea if it would be feasible or not. I think at least a comment about the "magic" indexes would be helpful. The test is failing on Python 3.